e2e: fixing hardcode values and formatted text

mulatinho opened the pull request:
#### What this PR does / why we need it:

the purpose of this PR is to make sure the messages printed in the output are equal to the gracePeriod set by the code, also, it does a refactor in some strings and values of the keys based on what is set on the code.

```sh
$ git diff HEAD~1 --stat
 test/e2e/network/pod_lifecycle.go | 39 ++++++++++++++++++++-------------------
 test/e2e/network/service.go       | 43 ++++++++++++++++++++++++-------------------
 2 files changed, 44 insertions(+), 38 deletions(-)
```

/cc @royalsflush

mulatinho commented:
/cc @aojea 
/assign @aojea 

hope it helps and makes sense for you :-)

royalsflush commented on test/e2e/network/pod_lifecycle.go:
```diff
@@ -45,9 +45,13 @@ var _ = common.SIGDescribe("Connectivity Pod Lifecycle", func() {
+		networkGracePeriodOfOneHundredSeconds int64 = 100
+		networkGracePeriodOfThirtySeconds     int64 = 30
+		networkHttpPort                       int   = 80
+		decimalFormat                         int   = 10
```
Two comments for these:
1. Why not make these local constants? It seems like they're used thorough the file?
2. Could you make the name agnostic to the values? Perhaps shortNetworkGracePeriod / longNetworkGracePeriod?
3. Nit: No need to add "decimal format", you can just use fmt.Sprintf for printing that as well
4. Nit: you can shorten the name (which is normally golang best practices, see https://google.github.io/styleguide/go/decisions) to httpPort

mulatinho replied:
@royalsflush I did the changes you asked, I also thought in use them as constants but we cannot get the address of a constants in the lines we do this, for example:

```go
webserverPod.Spec.TerminationGracePeriodSeconds = &longNetworkGracePeriod
```
`invalid operation: cannot take address of longNetworkGracePeriod (constant 100 of type int64)`

waiting for your feedback :-)

royalsflush replied:
It does, yes! Thanks for the changes

royalsflush commented:
/lgtm

(Be mindful you still need approval)

aojea commented on test/e2e/network/pod_lifecycle.go:
```diff
@@ -81,19 +83,18 @@ var _ = common.SIGDescribe("Connectivity Pod Lifecycle", func() {
-		gracePeriod := int64(100)
-		webserverPod := e2epod.NewAgnhostPod(ns, "webserver-pod", nil, nil, nil, "netexec", "--http-port=80", fmt.Sprintf("--delay-shutdown=%d", gracePeriod))
+		ginkgo.By("Creating 1 webserver pod able to serve traffic during the grace period of " + strconv.FormatInt(longNetworkGracePeriod, 10) + " seconds")
+		webserverPod := e2epod.NewAgnhostPod(ns, "webserver-pod", nil, nil, nil, "netexec", fmt.Sprintf("--http-port=%d --delay-shutdown=%d", httpPort, longNetworkGracePeriod))
```
this has to be comma separated, do not aggregate them

mulatinho replied:
ok, I thought fmt.Sprintf would make it slower two times. as aggregation seems faster IMHO. thanks for the feedback, done!

aojea commented on test/e2e/network/pod_lifecycle.go:
```diff
@@ -45,9 +45,12 @@ var _ = common.SIGDescribe("Connectivity Pod Lifecycle", func() {
+		cs                      clientset.Interface
+		ns                      string
+		podClient               *e2epod.PodClient
+		longNetworkGracePeriod  int64 = 100
+		shortNetworkGracePeriod int64 = 30
```
keeps the variables per test,

aojea commented on test/e2e/network/pod_lifecycle.go:
```diff
@@ -68,7 +70,7 @@ var _ = common.SIGDescribe("Connectivity Pod Lifecycle", func() {
 		ginkgo.By("Try to connect to the webserver")
 		// Wait until we are able to connect to the Pod
-		podIPAddress := net.JoinHostPort(webserverPod.Status.PodIP, strconv.Itoa(80))
+		podIPAddress := net.JoinHostPort(webserverPod.Status.PodIP, strconv.FormatInt(longNetworkGracePeriod, 10))
```
you are replacing a port by a graceperiod, the only thing neede to change here is to put 80 in a variable and use that

mulatinho replied:
oops, :/ sorry, fixed!

aojea commented on test/e2e/network/pod_lifecycle.go:
```diff
@@ -81,19 +83,18 @@ var _ = common.SIGDescribe("Connectivity Pod Lifecycle", func() {
 		ginkgo.By("Creating 1 client pod that will try to connect to the webservers")
-		pausePod := e2epod.NewAgnhostPod(ns, "pause-pod-1", nil, nil, nil, "netexec", "--http-port=80", fmt.Sprintf("--delay-shutdown=%d", gracePeriod))
-		pausePod.Spec.TerminationGracePeriodSeconds = &gracePeriod
+		pausePod := e2epod.NewAgnhostPod(ns, "pause-pod-1", nil, nil, nil, "netexec", fmt.Sprintf("--http-port=%d --delay-shutdown=%d", httpPort, longNetworkGracePeriod))
```
ditto https://github.com/kubernetes/kubernetes/pull/131571/files#r2071532333

aojea commented:
/hold

thanks for the contribution, but refactors need to be adding clear value, replacing hardcoded values by a common variable is, but I want to keep them per test.

Use different commits for changes and different files, do not mix unrelated things in the same commit, do not change things like comma separated arguments , ...

mulatinho commented:
> /hold
> 
> thanks for the contribution, but refactors need to be adding clear value, replacing hardcoded values by a common variable is, but I want to keep them per test.
> 
> Use different commits for changes and different files, do not mix unrelated things in the same commit, do not change things like comma separated arguments , ...


@aojea I did the changes, also put one commit per file , I think the change is useful because avoid future mistakes replacing constants instead of variables, but its your call, I hope you can accept the changes as its my first here :-) have a great weekend.

mulatinho commented:
/kind testing

aojea commented on test/e2e/network/pod_lifecycle.go:
```diff
@@ -81,19 +83,18 @@ var _ = common.SIGDescribe("Connectivity Pod Lifecycle", func() {
 
 	ginkgo.It("should be able to connect to other Pod from a terminating Pod", func(ctx context.Context) {
-		ginkgo.By("Creating 1 webserver pod able to serve traffic during the grace period of 300 seconds")
-		gracePeriod := int64(100)
```
leave this as it was per test, no need to consolidate them, also fix the comment that says 300 seconds, don;t change to `longNetworkGracePeriod`

mulatinho replied:
Yes, that was one of the main purposes of this PR, there is this inconsistency due to hardcoded values, thanks for understanding. I will do the changes.

aojea commented on test/e2e/network/pod_lifecycle.go:
```diff
@@ -81,19 +83,18 @@ var _ = common.SIGDescribe("Connectivity Pod Lifecycle", func() {
 
 		ginkgo.By("Creating 1 client pod that will try to connect to the webservers")
-		pausePod := e2epod.NewAgnhostPod(ns, "pause-pod-1", nil, nil, nil, "netexec", "--http-port=80", fmt.Sprintf("--delay-shutdown=%d", gracePeriod))
-		pausePod.Spec.TerminationGracePeriodSeconds = &gracePeriod
```
keep it

aojea commented on test/e2e/network/pod_lifecycle.go:
```diff
@@ -45,9 +45,12 @@ var _ = common.SIGDescribe("Connectivity Pod Lifecycle", func() {
+		cs                      clientset.Interface
+		ns                      string
+		podClient               *e2epod.PodClient
+		longNetworkGracePeriod  int64 = 100
+		shortNetworkGracePeriod int64 = 30
+		httpPort                int   = 80
```
don't expose this variables and leave them per test

aojea commented on test/e2e/network/pod_lifecycle.go:
```diff
@@ -152,7 +152,7 @@ var _ = common.SIGDescribe("Connectivity Pod Lifecycle", func() {
 		ginkgo.By("Creating 1 client pod that will try to connect to the blue-green-svc")
 		clientPod := e2epod.NewAgnhostPod(ns, "client-pod-1", nil, nil, nil)
-		clientPod.Spec.TerminationGracePeriodSeconds = &gracePeriod
+		clientPod.Spec.TerminationGracePeriodSeconds = &shortNetworkGracePeriod
```
leave it as it was

aojea commented on test/e2e/network/service.go:
```diff
@@ -1939,7 +1939,7 @@ var _ = common.SIGDescribe("Services", func() {
 						Name:  "agnhost",
 						Image: imageutils.GetE2EImage(imageutils.Agnhost),
-						Args:  []string{"netexec", "--http-port=80", fmt.Sprintf("--delay-shutdown=%d", gracePeriod)},
+						Args:  []string{"netexec", fmt.Sprintf("--http-port=%d", servicePort), fmt.Sprintf("--delay-shutdown=%d", gracePeriodSeconds)},
```
this is really container port, and this requires then to change 

> 				{Port: 80, Name: "http", Protocol: v1.ProtocolTCP, TargetPort: intstr.FromInt32(80)},

to Port: servicePort and TargetPort: containerPort

and the lines below that use 80 too

mulatinho replied:
I will fix all these lines to use `servicePort` and `containerPort` as well

mulatinho commented:
@aojea I reviewed all changes and I think now its bringing value as you requested, I fixed the exposition of variables, formatted the output lines and seems good to me, consistent. waiting for your feedback, hope it get approved :)

mulatinho commented:
/kind cleanup

aojea commented:
thanks for the effort but now you have some commits that go back and forth, you add some variables and then you remove then, can you have a clean history with the actual changes we want to merge?

mulatinho committed:
e2e: service: removing hardcoded values to vars

this commit removes hardcoded constants and turn them into variables and
format some messages through the code using fmt.Sprint

Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

mulatinho committed:
e2e: pod_lifecycle: removing hardcoded values to vars

this commit removes hardcoded constants and turn them into variables and
format some messages through the code using fmt.Sprint

Signed-off-by: Alexandre Mulatinho <alex@mulatinho.net>

mulatinho commented:
/assign @danwinship 
/assign @andrewsykim

aroradaman commented:
/test pull-kubernetes-e2e-kind

mulatinho commented:
> /test pull-kubernetes-e2e-kind

thanks brother, I think it will need approval first ./cc @aroradaman

aroradaman commented:
/ok-to-test

mulatinho commented:
/retest-required
